app/controllers/provider_interface/application_choices_controller.rb

```diff
@@ -22,8 +22,8 @@ def show
     # by a proper ProviderUser when implementing Signin.
     def current_user
       fake_user_class = Struct.new(:provider)
-      fake_provider_class = Struct.new(:code)
-      fake_user_class.new(fake_provider_class.new('ABC'))
+      fake_provider = Provider.find_or_create_by(code: 'ABC')
```

Comment:
I think shouldn't create test data like this anymore now that we're "live", since it could pollute the production database.

Reply:
I feel like test stubbing should not be in this controller at all and we need a nicer what to stub out the current_user in tests.

Reply:
Updated the stub to only find and not create.  
here: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/479/commits/77c895fac48fc37fc154ac653b0918f6061690ed

I would like to move that stub out of the controller, I can talk to @duncanjbrown about his current PR around DFE Sign In https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/507

Reply:
Propose keeping the change `find_by` for now, so it works in development and breaks in prod, and we'll have Sign-in fixed up in the near future? Will all be behind basic auth anyway

Reply:
(Sign in vars are not configured on prod rn, so if we deployed #507 the effect of clicking "Sign in with DfE Sign-in" is 💥 )

Reply:
Sounds like a good approach for now. 👍